torch/lib/THD/master_worker/master/State.hpp

```diff
@@ -1,18 +1,26 @@
 #pragma once
 
+#include "../../base/DataChannel.h"
+#include "../../base/ChannelUtils.hpp"
+
 #include <cstddef>
 #include <cstdint>
+#include <memory>
+#include <mutex>
 #include <vector>
 
 namespace thd {
 namespace master {
 
 struct WorkerState {
+  WorkerState() : mutex(std::make_shared<std::mutex>()) {}
+
+  std::shared_ptr<std::mutex> mutex;
```

Comment:
Because we use copy constructor here: https://github.com/apaszke/pytorch-dist/blob/86070802b80c26b45185fc270729d09ca2d32e14/torch/lib/THD/master_worker/master/Master.cpp#L31